src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs

```diff
@@ -49,19 +49,38 @@ await foreach (var ar in analyzeResultDefinitions)
                         {
                             Name = ar.Name,
                             SemanticVersion = ar.Version,
-                            InformationUri = ar.InformationURI,
-                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(a => new ReportingDescriptor()
-                            {
-                                Id = a.Key,
-                                FullDescription = new() { Text = a.First().RuleName, },
-                            }).ToList(),
+                            InformationUri = ar.InformationUri,
+                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(ExtractRule).ToList(),
                         },
                     },
                     Results = ExtractResults(analyzeResults),
                 };
             }
         }
 
+        private static ReportingDescriptor ExtractRule(IGrouping<string, AnalyzeResult> analyzeResults)
+        {
+            var analyzeResult = analyzeResults.First();
```

Comment:
Why only `.First()`? We do need all the distinct rules to be assigned to `Rules`.

Reply:
this method would receive the group itself, so, it's already grouped. Check line 53.

With that, all elements have the same ruleId, so, we can just get the  first since all metadata will be the same.

Reply:
Have you tested this with scenarios where multiple distinct rules are present like in case of `API Upgradability` section?

Reply:
Just updated a test to verify.